.github/scripts/assign_reviewer.py

```diff
@@ -0,0 +1,311 @@
+        f"Could not fetch PR author for PR {pr_number}"
+    )
+    return result.stdout.strip()
+
+
+def calculate_current_reviewer(reviewers: List[str], rotation_info: RotationInfo, exclude_user: Optional[str] = None) -> Optional[str]:
+    """Calculate the current reviewer based on the rotation schedule, excluding specified user."""
+    if not reviewers:
+        print("Error: No reviewers found")
+        return None
+    
+    if rotation_info["before_start"]:
+        print(f"Warning: Current date is before start date. Using first reviewer.")
+        # Find first reviewer that's not the excluded user
+        for reviewer in reviewers:
+            if reviewer != exclude_user:
+                return reviewer
+        return reviewers[0] if reviewers else None
+    
+    # Calculate total weeks since start and map to reviewer
+    # Each reviewer gets rotation_cycle_weeks weeks, then we cycle to the next
+    total_weeks = int(rotation_info["weeks_since_start"])
+    rotation_cycle_weeks = rotation_info["rotation_cycle_weeks"]
+    reviewer_index = (total_weeks // rotation_cycle_weeks) % len(reviewers)
+    
+    # If the calculated reviewer is the excluded user, find the next one
+    if exclude_user and reviewers[reviewer_index] == exclude_user:
+        # Try next reviewer in the list
+        next_reviewer_index = (reviewer_index + 1) % len(reviewers)
+        attempts = 0
+        while reviewers[next_reviewer_index] == exclude_user and attempts < len(reviewers):
+            next_reviewer_index = (next_reviewer_index + 1) % len(reviewers)
+            attempts += 1
+        
+        # If all reviewers are excluded, return None
+        if reviewers[next_reviewer_index] == exclude_user:
+            print(f"Warning: All reviewers are excluded ({exclude_user}), skipping assignment")
+            return None
+        
+        return reviewers[next_reviewer_index]
+    
+    return reviewers[reviewer_index]
+
+
+def run_gh_command(args: List[str], error_message: str) -> subprocess.CompletedProcess:
+    """Run a GitHub CLI command with consistent error handling."""
+    try:
+        return subprocess.run(
+            ['gh'] + args,
+            capture_output=True, text=True, check=True
+        )
+    except FileNotFoundError:
+        print("Error: 'gh' command not found. Is the GitHub CLI installed and in the PATH?")
+        sys.exit(1)
+    except subprocess.CalledProcessError as e:
+        print(f"{error_message}: {e.stderr}", file=sys.stderr)
+        sys.exit(1)
+
+
+def get_existing_reviewers(pr_number: str) -> List[str]:
+    """Get list of reviewers already assigned to the PR."""
+    repo = os.environ.get('GITHUB_REPOSITORY', 'bootc-dev/bootc')
+    result = run_gh_command(
+        ['api', f'repos/{repo}/pulls/{pr_number}', '--jq', '.requested_reviewers[].login'],
+        f"Could not fetch existing reviewers for PR {pr_number}"
+    )
+    return result.stdout.strip().split('\n') if result.stdout.strip() else []
```

Comment:
Not that it matters at all but just skimming this stood out to me, it's  inefficient to allocate a string, trim it (allocating a whole new copy) and then re-trim again and allocate *again* but anyways, it doesn't matter